cmd/lifecycle/analyzer.go

```diff
@@ -96,26 +95,21 @@ func (a *analyzeCmd) Privileges() error {
 
 // Exec executes the command.
 func (a *analyzeCmd) Exec() error {
-	factory := phase.NewAnalyzerFactory(
+	factory := phase.NewConnectedFactory(
 		a.PlatformAPI,
 		&cmd.BuildpackAPIVerifier{},
 		NewCacheHandler(a.keychain),
-		phase.NewConfigHandler(),
+		phase.Config,
 		image.NewHandler(a.docker, a.keychain, a.LayoutDir, a.UseLayout, a.InsecureRegistries),
 		image.NewRegistryHandler(a.keychain, a.InsecureRegistries),
 	)
-	analyzer, err := factory.NewAnalyzer(a.AdditionalTags, a.CacheImageRef, a.LaunchCacheDir, a.LayersDir, a.OutputImageRef, a.PreviousImageRef, a.RunImageRef, a.SkipLayers, cmd.DefaultLogger)
+	analyzer, err := factory.NewAnalyzer(*a.LifecycleInputs, cmd.DefaultLogger)
```

Comment:
WDYT about something like this to make it read better?
```suggestion
	analyzer, err := factory.NewAnalyzer(a.Inputs(), cmd.DefaultLogger)
```

It might also give us a chance to do more in the future if needed

```golang
	analyzer, err := factory.NewAnalyzer(a.Inputs(WithSomeOption(blah)), cmd.DefaultLogger)
```

Reply:
Added in ab3a3da5